src/libraries/System.Private.CoreLib/src/Resources/Strings.resx

```diff
@@ -1110,15 +1158,21 @@
   <data name="Argument_InvalidArrayLength" xml:space="preserve">
     <value>Length of the array must be {0}.</value>
   </data>
-  <data name="Argument_InvalidArrayType" xml:space="preserve">
-    <value>Target array type is not compatible with the type of items in the collection.</value>
```

Comment:
The existing uses of `Argument_InvalidArrayType` use the message for situations where the array type is valid, but not compatible with the collection. `Array type is invalid.` is not appropriate for existing uses. Could you please rename the existing `Argument_InvalidArrayType` to `Argument_IncompatibleArrayType`, and use `Argument_InvalidArrayType` only in the case where the array type is really invalid.